internal/ledger/store/badger_test.go

```diff
@@ -97,3 +97,81 @@ func TestBadgerDB_Set(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, expectedValue, value)
 }
+
+func TestBadgerDB_NewIterator(t *testing.T) {
+	t.Parallel()
+
+	tmpdir, err := ioutil.TempDir("", "bdb-test-")
+	defer os.RemoveAll(tmpdir)
+	assert.NoError(t, err)
+
+	db, err := NewBadgerDB(tmpdir)
+	require.NoError(t, err)
+
+	var (
+		commonScope      Scope
+		commonPrefix     []byte
+		randKeys         []testBadgerKey
+		unexpectedValues [][]byte
+		keys             []testBadgerKey
+		expectedValues   [][]byte
+		expected         map[string][]byte
+	)
+
+	expected = make(map[string][]byte)
```

Comment:
It's also important to assert the ordering of items (it will help us implement sequence).
I propose we make a sorted list of kv pairs. `results := []kv`.

Reply:
done